examples/hotrod/pkg/tracing/http.go

```diff
@@ -21,17 +21,44 @@ import (
 	"errors"
 	"io"
 	"net/http"
+	"net/http/httptrace"
+	"net/url"
 
-	"github.com/opentracing-contrib/go-stdlib/nethttp"
-	"github.com/opentracing/opentracing-go"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type contextKey int
+
+const (
+	keyTracer contextKey = iota
 )
 
 // HTTPClient wraps an http.Client with tracing instrumentation.
 type HTTPClient struct {
-	Tracer opentracing.Tracer
+	Tracer trace.TracerProvider
 	Client *http.Client
 }
 
+// Tracer holds tracing details for one HTTP request.
+type Tracer struct {
```

Comment:
Isn't there std http instrumentation in OTEL similar to grpc interceptors?

Reply:
https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp